src/ca/mcgill/cs/jetuml/application/JetExtensions.java

```diff
@@ -0,0 +1,73 @@
+package ca.mcgill.cs.jetuml.application;
+
+import java.util.LinkedList;
+import java.util.List;
+import java.util.ResourceBundle;
+
+import ca.mcgill.cs.jetuml.UMLEditor;
+import javafx.stage.FileChooser.ExtensionFilter;
+
+
+/**
+ * A class to create and access diagram extension filters 
+ * used by the file chooser.
+ * 
+ * @author Kaylee I. Kutschera
+ */
+public final class JetExtensions 

```

Comment:
* I would rename `FileExtensions` to distinguish from other potential types of extensions. 
* This class does not need to be a singleton: it can be a utility class with two static methods `getAll()` and `get(String)`. This will reduce the amount of state to track in the window objects.
* The tests in `TestEditorFrame` are broken: they should be ported to this new design.